libnmstate/netapplier.py

```diff
@@ -164,6 +165,20 @@ def sanitize_ethernet_state(ifaces_desired_state, ifaces_current_state):
     return ifaces_desired_state
 
 
+def sanitize_dhcp_state(ifaces_desired_state):
+    """
+    If DHCP enabled, if desired state does not contain any IP address,
```

Comment:
What about autoconf?
Perhaps it should be something like this:
`When dynamic IP is enabled and the address missing, set an empty address list. This assures that the desired state is not complemented by the current state address values.`